DMCompiler/Bytecode/DreamProcOpcode.cs

```diff
@@ -402,3 +402,50 @@ public static class OpcodeVerifier {
         return BitConverter.ToString(hashBytes).Replace("-", "");
     }
 }
+
+/// <summary>
+/// Custom attribute for declaring <see cref="OpcodeMetadata"/> metadata for individual opcodes
+/// </summary>
+[AttributeUsage(AttributeTargets.Field, Inherited = false, AllowMultiple = false)]
+internal sealed class OpcodeMetadataAttribute : Attribute {
+    public OpcodeMetadata Metadata;
+
+    public OpcodeMetadataAttribute(int stackDelta = 0) {
+        Metadata = new OpcodeMetadata(stackDelta);
+    }
+}
+
+/// <summary>
+/// Miscellaneous metadata associated with individual <see cref="DreamProcOpcode"/> opcodes using the <see cref="OpcodeMetadataAttribute"/> attribute
+/// </summary>
+public struct OpcodeMetadata {
+    public readonly int StackDelta; // Net change in stack size caused by this opcode
+
+    public OpcodeMetadata(int stackDelta = 0) {
+        StackDelta = stackDelta;
+    }
+}
+
+/// <summary>
+/// Automatically builds a cache of the <see cref="OpcodeMetadata"/> attribute for each opcode
+/// </summary>
+public static class OpcodeMetadataCache
+{
+    private static readonly OpcodeMetadata[] MetadataCache = new OpcodeMetadata[256];
+
+    static OpcodeMetadataCache()
+    {
+        foreach (DreamProcOpcode opcode in Enum.GetValues(typeof(DreamProcOpcode)))
+        {
+            var field = typeof(DreamProcOpcode).GetField(opcode.ToString());
+            var attribute = Attribute.GetCustomAttribute(field!, typeof(OpcodeMetadataAttribute));
+            var metadataAttribute = Unsafe.As<OpcodeMetadataAttribute>(attribute);
```

Comment:
Why the `Unsafe.As<>()`? a normal cast should work fine

Reply:
Why not?

Reply:
It's generally good to stay away from unsafe things unless necessary.

While `attribute` will probably always be an `OpcodeMetadataAttribute` here, you're still foregoing any safety checks which could lead to really odd issues down the line.

Reply:
```suggestion
            var metadataAttribute = (OpcodeMetadataAttribute)attribute;
```